pro_wes/ga4gh/wes/workflow_runs.py

```diff
@@ -96,13 +96,22 @@ def run_workflow(
         # write workflow attachments
         self._save_attachments(attachments=document_stored.attachments)
 
+        if document_stored.wes_endpoint is None:
+            raise ValueError("No WES endpoint available.")
+
+        if document_stored.wes_endpoint.base_path is None:
+            raise ValueError("No WES endpoint base path provided.")
+
         # instantiate WES client
         wes_client: WesClient = WesClient(
             host=document_stored.wes_endpoint.host,
             base_path=document_stored.wes_endpoint.base_path,
             token=kwargs.get("jwt", None),
         )
 
+        if document_stored.task_id is None:
+            raise ValueError("No task ID available")
+
```

Comment:
Hmm, I see that the type for `document_stored` is not indicated. If you add a type hint for that, do we still need these `None` checks for mypy?

And if so, perhaps it's worth adding one or more custom exceptions (or, if applicable, reuse the custom exceptions already defined)? Custom exceptions are defined in `foca.exceptions`, i.e., [here](https://github.com/elixir-cloud-aai/proWES/blob/dev/pro_wes/exceptions.py)). Please note that if you define custom exceptions, you also need to include them in the `exceptions` dictionary in the same module, and add the desired error code and message to be returned to the user as a JSON object.

Reply:
Just FYI: Any error that is raised and that is _not_ listed specifically in the `exceptions` dictionary (e.g., `ValueError`) will be automatically routed to `InternalServerError`, with a `500` response code and a generic, uninformative `An unexpected error occurred` message that isn't overly helpful to the user. Hence, custom errors are to be preferred.